### PR TITLE
Implement cleanup of temporary files after execution

### LINKED ISSUE #59: Cleanup of temporary helmfiles
Hi,

Great tool! very happy to have this available, and we're going to start using it a lot.

One thing I've noticed, is that there are temporary files left in the `working_dir` after the plan is applied - should I be expecting these to be cleaned up automatically? or should I be adding them to `.gitignore` or something?

example:
```
❯❯❯❯ ll
total 328
drwxr-xr-x  35 andrew  staff   1.1K Apr  1 15:46 ./
drwxr-xr-x   5 andrew  staff   160B Mar 30 07:26 ../
drwxr-xr-x   6 andrew  staff   192B Mar 26 14:30 .terraform/
-rw-r--r--   1 andrew  staff   5.3K Mar 29 16:11 .terraform.lock.hcl
-rw-r--r--   1 andrew  staff    17B Mar 26 15:03 .tool-versions
-rw-r--r--   1 andrew  staff   1.2K Mar 23 12:24 2048_full.yaml
-rw-r--r--   1 andrew  staff   222B Mar 26 14:13 backend.tf
-rw-r--r--   1 andrew  staff   782B Mar 24 11:39 cluster.tf
-rw-r--r--   1 andrew  staff   1.9K Mar 23 13:48 codepipeline.tf
-rw-r--r--   1 andrew  staff   1.6K Apr  1 06:31 data-resources.tf
-rw-r--r--   1 andrew  staff   1.5K Mar 29 09:09 eks-fargate-profile.tf
-rw-r--r--   1 andrew  staff    41K Apr  1 09:23 errored.tfstate
drwxr-xr-x   3 andrew  staff    96B Mar 23 18:19 files/
-rw-r--r--   1 andrew  staff   556B Mar 25 14:41 helm-alb.tf
-rw-r--r--   1 andrew  staff   1.1K Apr  1 14:07 helm-hello-world.tf
-rw-r--r--   1 andrew  staff   1.3K Apr  1 13:52 helm-ingress-nginx.tf
-rw-r--r--   1 andrew  staff   2.1K Mar 30 08:14 helm-rbac-manager.tf
-rwxr-xr-x   1 andrew  staff   358B Apr  1 13:57 helmfile-601c16fc8dec00a95b32d832dcf9a628f9fc92e39cc06148f02417884b4adb00.yaml*
-rwxr-xr-x   1 andrew  staff   357B Apr  1 13:57 helmfile-7ea24fb586ca43355e166e8b17927e7c571ce14fb7e30901f7ce0faa89336325.yaml*
-rwxr-xr-x   1 andrew  staff   603B Apr  1 14:16 helmfile-8bfc9a1633dc2affbd39cf2601b0d2c04ef11f8fa7178b7eb6a542dae252f7c6.yaml*
-rwxr-xr-x   1 andrew  staff   418B Apr  1 14:16 helmfile-a46944bfce1d865911b70bd8a6d0e95fdd83f3b7695de04e1b6d5c5b2d490377.yaml*
-rwxr-xr-x   1 andrew  staff   420B Apr  1 14:16 helmfile-bcdaf754eea3d447ebd765fe8436885957a04b942920d2d0e14a765f86ea3080.yaml*
-rwxr-xr-x   1 andrew  staff   242B Apr  1 14:16 helmfile-c359764e1d8ab998de6ed447b29f328ba1aab2df71f32c7dfb7be37d5c5c235f.yaml*
-rwxr-xr-x   1 andrew  staff   417B Apr  1 14:17 helmfile-c4fde063873396efb34d44ff636e9f9752ff9299f2a013d70435119f692d00d3.yaml*
-rwxr-xr-x   1 andrew  staff   1.5K Apr  1 14:16 helmfile-cd4891323a42f055a1ff2162f98a7dcf23589d47be26785715c2a333a5525e6a.yaml*
-rwxr-xr-x   1 andrew  staff   419B Apr  1 14:17 helmfile-cd79c8f04c3a0b4633980087ff8015989efc992e10382e14c1d657b67104dc24.yaml*
-rwxr-xr-x   1 andrew  staff   714B Apr  1 14:16 helmfile-de5e90dc18fa599b32333394ca490f915d3b860d712fdff263ab7163f95bbf09.yaml*
-rw-r--r--   1 andrew  staff   394B Mar 30 08:54 k8s.tf
-rw-r--r--   1 andrew  staff   414B Mar 23 18:34 logging.tf
-rw-r--r--   1 andrew  staff   1.1K Apr  1 12:25 providers.tf
-rw-r--r--   1 andrew  staff   384B Mar 29 18:51 rbac.tf
-rw-r--r--   1 andrew  staff   422B Mar 25 14:11 rfc.tf
drwxr-xr-x   9 andrew  staff   288B Apr  1 14:16 templates/
-rw-r--r--   1 andrew  staff   392B Mar 26 16:03 values.yaml
-rw-r--r--   1 andrew  staff   3.8K Mar 29 10:14 variables.tf

```

### PR DESCRIPTION
**Pull request by dm3ch**
Implement cleanup of temporary helmfiles after execution

/fix https://github.com/mumoshu/terraform-provider-helmfile/issues/59

### CODE REVIEW

**Comment by dm3ch** [comment]
@mumoshu PR is ready for review.

Each change and it's reason described in commit messages

**Comment by dm3ch** [comment]
Any chance to get it merged?

File: pkg/helmfile/release_set.go [hunk 1]
```diff
@@ -69,12 +69,6 @@ func NewReleaseSet(d ResourceRead) (*ReleaseSet, error) {
 	if env := d.Get(KeyEnvironment); env != nil {
 		f.Environment = env.(string)
 	}
-	// environment defaults to "" for helmfile_release_set but it's always nil for helmfile_release.
-	// This nil-check is required to handle the latter case. Otherwise it ends up with:
-	//   panic: interface conversion: interface {} is nil, not string
-	if path := d.Get(KeyPath); path != nil {
-		f.Path = path.(string)
-	}
```
Comment (mumoshu) [inline pkg/helmfile/release_set.go:77]: I'm going to merge this and make necessary changes(if any) later but one thing to ask here- What was the rationale behind removing this feature (specifying an existing path) in this pull request?

**Review by mumoshu — APPROVED** [review]
Mostly LGTM with only one question! Thanks for your contribution @dm3ch

File: pkg/helmfile/release_set.go (hunk 1 above)
Comment (dm3ch) [reply pkg/helmfile/release_set.go:77]: @mumoshu When I started finding root cause of persisting tmp files, i found that there's two separate pieces of code creating tmp files for releases:
1) Creation of helmfile for standalone helm release
2) Copying existing helmfile before apply

This cases had completely different places where tmp files were created, so I refactored both cases to pass helmfile content same way and make same tmp file creation code for them.

When I sent this PR I tested that both terraform resources works fine with this PR, so my idea was unification of tmp file creation without breaking how terraform resources works

**Comment by yuriipolishchuk** [comment]
@mumoshu may we have a new release with this improvement, please?

@dm3ch thank you 👍

**Pull request merged**